### DISCUSSION TITLE
Telego with webhook over ngrok

### DISCUSSION THREAD

**Discussion by abakum** [mymmrac/telego#107] [Q&A] [answered]
I successfully used Telego with webhook over ngrok.exe client: https://github.com/abakum/tmbPing/blob/12f2388ff8bf778a1e18807f9850743e2f910967/main.go#L149
but have use it over go library like https://github.com/ngrok/ngrok-go/issues/65
?

**Comment by abakum** [comment]
It works:
https://github.com/abakum/tmbPing/blob/37bde72f7746a2bf017f8b1e9b804a61c16393df/main.go#L186
Maybe it can be done better?

**Reply by mymmrac** [reply]
Hmm, actually I never worked with ngrok lib, I will check it out and will get back with response once I will have more understanding of it

**Reply by mymmrac** [reply, marked as answer]
I added an example of Ngrok usage https://github.com/mymmrac/telego/blob/main/examples/ngrok/main.go

Added `FuncWebhookServer` to simplify creation, generally it's similar to what you showed, but much simpler - using `Serve` method of server to avoid dealing with each connection separately.

I also want to add some utility methods to simplify webhook server creation, so it will be even simpler after that.
